src/control/server/server.go

```diff
@@ -249,9 +250,20 @@ func Start(log *logging.LeveledLogger, cfg *Configuration) error {
 	go func() {
 		sig := <-sigChan
 		log.Debugf("Caught signal: %s", sig)
-		if err := drpcCleanup(cfg.SocketDir); err != nil {
-			log.Errorf("error during dRPC cleanup: %s", err)
+
+		// Attampt graceful shutdown of I/O servers.
+		timeout, err := time.ParseDuration(cfg.ShutdownTimeout)
+		if err != nil {
+			log.Errorf("parsing shutdown timeout from config: %s", err)
+		} else {
```

Comment:
the value has already been parsed at this point in config validation, I was originally going to ignore any error here ("timeout, _ :=") but thought that probably wouldn't be acceptable to you. would you like me to ignore the error, store the parsed value on validation, or something else?

Reply:
Your assertion is true for the code as it is currently written, but it's conceivable that the code could change so that it's possible to get here without having gone through that validation path. If the timeout will continue to be configurable, then it would be preferable to parse and handle errors outside of the signal handler so that server startup fails due to the bad value.